packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -620,7 +620,12 @@ AndroidEmbeddingVersion _getAndroidEmbeddingVersion(FlutterProject project) {
 Future<void> _writeAndroidPluginRegistrant(FlutterProject project, List<Plugin> plugins) async {
   final List<Map<String, dynamic>> androidPlugins =
```

Comment:
How about instead of sorting the untyped map, create a new List of plugins here and apply the sort on the typed Plugin objects:

```
final List<Plugin> ordered = plugins.toList()
  ..sort(...);
```

Reply:
I don't know if I undestane but I will take a look and update here when I can!

Reply:
@jonahwilliams i checked it out and this does not make sense because _extractPlatformMaps also removes all plugins not related to Android and i can only "remove" the others inside the _writeAndroidPluginRegistrant function.  
For example:

```
url_launcher - {name: url_launcher, package: io.flutter.plugins.urllauncher, class: UrlLauncherPlugin, supportsEmbeddingV1: true, supportsEmbeddingV2: true}
url_launcher_linux - null
url_launcher_macos - null
url_launcher_web - null
url_launcher_windows - null
video_player - {name: video_player, package: io.flutter.plugins.videoplayer, class: VideoPlayerPlugin, supportsEmbeddingV1: true, supportsEmbeddingV2: true}
video_player_web - null
```

Also, I Only know if the plugin is V2 or not inside the "toMap" function.

  ```
@override
  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      'package': package,
      'class': pluginClass,
      // Mustache doesn't support complex types.
      'supportsEmbeddingV1': _supportedEmbedings.contains('1'),
      'supportsEmbeddingV2': _supportedEmbedings.contains('2'),
    };
  }
```

What I could do  is to expose a new method in the AndroidPlugin class that would tell me if the plugin is V2 or not.

But then i would need to check  if the plugin is Android(which is done inside the `_extractPlatformMaps`) and ifit is V2.

IMHO we should use the Untyped map instead of double checking twice the platform.

LMK what you think!